test/integration/integration_test.go

```diff
@@ -957,14 +949,140 @@ func TestDockerfileBuildSourceScriptsRun(t *testing.T) {
 		t.Errorf("Unable to create run file: %v", err)
 	}
 
+	config := &api.Config{
+		BuilderImage: "docker.io/centos/nodejs-8-centos7",
+		AssembleUser: "",
+		ImageWorkDir: "",
+		Source:       git.MustParse("file:///" + filepath.ToSlash(sourcecode)),
+		ForceCopy:    true,
+		ScriptsURL:   "",
+		Injections:   api.VolumeList{},
+		Destination:  "/destination",
+
+		Environment: api.EnvironmentList{},
+		Labels:      map[string]string{},
+
+		AsDockerfile: filepath.Join(tempdir, "Dockerfile"),
+	}
+	expected := []string{
+		"COPY --chown=1001:0 upload/scripts /destination/scripts",
+		"RUN /usr/libexec/s2i/assemble",
+		"CMD /destination/scripts/run",
+	}
+	expectedFiles := []string{
+		filepath.Join(tempdir, "upload/scripts/run"),
+	}
+	runDockerfileTest(t, config, expected, nil, expectedFiles)
+}
+
+// TestDockerfileBuildScriptsURLImage tests the behavior if the ScriptsURL
+// is set to an image:// URL. In this case we blind trust that the image
+// contains all of the s2i scripts at the given directory, regardless
+// of what is contained in the source.
+func TestDockerfileBuildScriptsURLImage(t *testing.T) {
```

Comment:
Updated existing test - it now ensures that we don't upload scripts if ScriptsURL points to an image